webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/design/SiteDesignActionHandler.java

```diff
@@ -0,0 +1,660 @@
+ * "http://www.silverpeas.org/docs/core/legal/floss_exception.html"
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.silverpeas.components.websites.servlets.design;
+
+import org.silverpeas.components.websites.control.WebSiteSessionController;
+import org.silverpeas.components.websites.service.WebSitesException;
+import org.silverpeas.components.websites.servlets.WebSitesRequestRouter;
+import org.silverpeas.components.websites.siteManage.model.SiteDetail;
+import org.silverpeas.core.util.EncodeHelper;
+import org.silverpeas.core.util.ResourceLocator;
+import org.silverpeas.core.util.SettingBundle;
+import org.silverpeas.core.web.http.HttpRequest;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ *
+ * @author mmoquillon
+ */
+public class SiteDesignActionHandler {
+
+  private final WebSiteSessionController controller;
+
+  public SiteDesignActionHandler(final WebSiteSessionController controller) {
+    this.controller = controller;
+  }
+
+  public String handle(final HttpRequest request) throws WebSitesException {
+    // Action = newSite the first time, never null
+    String action = request.getParameter("Action");
+    String id = request.getParameter("Id"); // jamais null sauf en creation ou en
+    // Retrieve currentPath parameter : null when creating a webSite
+    String currentPath = request.getParameter("path");
+    if (currentPath != null) {
+      currentPath = doubleAntiSlash(currentPath);
+    }
+
+    // ADD NEW SITE -------------------------------------------------------------
+    if ("newSite".equals(action)) {
+      currentPath = addNewDesignedSite(controller, request);
+    } else if ("updateDescription".equals(action)) { // type 0 design ou 2 upload
+      updateDesignedSite(controller, request, currentPath);
+    } else if ("addFolder".equals(action)) {
+      addFolderInSite(controller, request, currentPath);
+    } else if ("renameFolder".equals(action)) {
+      renameFolderInSite(controller, request, currentPath);
+    } else if ("deleteFolder".equals(action)) {
+      deleteFolderInSite(controller, request, currentPath);
+    } else if ("addPage".equals(action)) {
+      addPageInSite(controller, request, currentPath);
+    } else if ("renamePage".equals(action)) {
+      renamePageInSite(controller, request, currentPath);
+    } else if ("deletePage".equals(action)) {
+      deletePageInSite(controller, request, currentPath);
+    } else if ("classifySite".equals(action)) { // cas de l'upload et du design
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ATrailingCommentCheck)